### PR TITLE
Replaces remaining developerErrors with Check if possible

### PR DESCRIPTION
**Pull request by hanbollar**
Fixes lots of classes for [#4794](https://github.com/AnalyticalGraphicsInc/cesium/issues/4794)

dont merge in yet - im still doing final checks

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Change Log
 * `BingMapsImageryProvider` is no longer the default base imagery layer.
 * `BingMapsGeocoderService` is no longer the default geocoder service.
 * Cesium no longer ships with a demo Bing Maps API key.
+* `ClippingPlaneCollection` now uses `ClippingPlane` objects instead of`Plane` objects. [#6498](https://github.com/AnalyticalGraphicsInc/cesium/pull/6498)
```
Comment (hpinkos) [inline CHANGES.md]: This change shouldn't be in this branch

File: Source/Core/AssociativeArray.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 define([
+        './Check',
         './defined',
         './defineProperties',
         './DeveloperError'
     ], function(
+        Check,
```
Comment (hpinkos) [inline Source/Core/AssociativeArray.js]: remove this if you didn't end up using it in `AssociativeArray`

File: Source/Core/CylinderGeometry.js [hunk 3]
```diff
@@ -78,18 +76,10 @@ define([
...
-        if (!defined(bottomRadius)) {
-            throw new DeveloperError('options.bottomRadius must be defined.');
-        }
-        if (slices < 3) {
-            throw new DeveloperError('options.slices must be greater than or equal to 3.');
-        }
+        Check.defined('length', length);
+        Check.defined('topRadius', topRadius);
+        Check.defined('bottomRadius', bottomRadius);
+        Check.number.greaterThanOrEquals('slices', slices, 3);
```
Comment (hpinkos) [inline Source/Core/CylinderGeometry.js]: Yeah, looks like you accidentally have some changes from your other branches here.  Do you need to merge in master?

File: Source/Core/Math.js [hunk 4]
```diff
@@ -410,6 +412,7 @@ define([
      */
     CesiumMath.toRadians = function(degrees) {
         //>>includeStart('debug', pragmas.debug);
+        Check.defined('degrees', degrees);
         if (!defined(degrees)) {
```
Comment (hpinkos) [inline Source/Core/Math.js]: remove this developer error

Comment (hpinkos) [reply Source/Core/Math.js]: Most of the variables in these math functions should be `Check.typeOf.number`

File: Source/Core/Math.js [hunk 5]
```diff
@@ -589,9 +572,7 @@ define([
      */
     CesiumMath.factorial = function(n) {
         //>>includeStart('debug', pragmas.debug);
-        if (typeof n !== 'number' || n < 0) {
-            throw new DeveloperError('A number greater than or equal to 0 is required.');
-        }
+        Check.number.greaterThanOrEquals('n', n, 0);
```
Comment (hpinkos) [inline Source/Core/Math.js]: `Check.typeOf.number.greaterThanOrEquals`

Comment (hpinkos) [reply Source/Core/Math.js]: same throughout

File: Source/Core/PolylineVolumeOutlineGeometry.js [hunk 6]
```diff
@@ -133,12 +135,8 @@ define([
         var shape = options.shapePositions;
 
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(positions)) {
-            throw new DeveloperError('options.polylinePositions is required.');
-        }
-        if (!defined(shape)) {
-            throw new DeveloperError('options.shapePositions is required.');
-        }
+        Check.defined('positions', positions);
```
Comment (hpinkos) [inline Source/Core/PolylineVolumeOutlineGeometry.js]: `Check.defined('options.polylinePositions', positions);`

The string should always be the param name passed in

File: Source/Core/QuaternionSpline.js [hunk 7]
```diff
@@ -60,12 +62,9 @@ define([
...
 
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(points) || !defined(times)) {
-            throw new DeveloperError('points and times are required.');
-        }
-        if (points.length < 2) {
-            throw new DeveloperError('points.length must be greater than or equal to 2.');
-        }
+        Check.defined('points', points);
+        Check.defined('times', times);
```
Comment (hpinkos) [inline Source/Core/QuaternionSpline.js]: revert these two.  Either `points` or `times` has to be defined

**Comment by hanbollar** [comment]
@hpinkos I'm still going through this but right now just as a question - `Check` isnt visible in the api documentation but `DeveloperError` is - Should I be changing all the instances of `DeveloperError` to Check (where appropriate) including Sandcastle examples and in the Documentation guide even though outside viewers wouldnt be able to look Check up in the API? or should I leave those just as `DeveloperError`. 

I think we should have those changed to Check if possible so that we dont have to switch all the new items merged in from dev error to check, but if there's a reason the Check API is still hidden currently I'll keep those as Dev error instead - lmk what u think

**Comment by hpinkos** [comment]
@hanbollar `Check` throws a `DeveloperError`, it's just a collection of private helper functions to make error messages consistent.  So we don't need to change the exception documentation where it says things like `@exception {DeveloperError} some param must be greater than 0`.  Is this what you meant?

I don't think we have any Sandcastle examples that throw `DevelperErrors` so I don't think there would be any changes there.

**Comment by hanbollar** [comment]
@hpinkos In the CustomDataSource.html Sandcastle example and in the DocumentationGuide itself when it's explaining exceptions (didnt mean actual documentation - miswording on my part)

**Comment by hpinkos** [comment]
Oh gotcha.  Yeah I would just leave those as they are

**Comment by hanbollar** [comment]
dont have enough time this summer to work on this as i had hoped. hopefully will be able to work more on this or other features later on.

**Comment by hpinkos** [comment]
Thanks @hanbollar, have a great summer!

**Pull request closed without merging**